src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs

```diff
@@ -25,11 +25,11 @@ public sealed class Continue
             private Continue() { }
         }
 
-        public static Props Props(long fromOffset, TimeSpan? refreshInterval, int maxBufferSize, string writeJournalPluginId)
+        public static Props Props(long fromOffset, TimeSpan? refreshInterval, int maxBufferSize, IActorRef writeJournal)
```

Comment:
I made this change on all of the `Query` implementations - for the simple reason that we're going to swap out the real underlying `JournalRef` with an `IActorRef` that points to the `SqlQueryThrottler` once this change is introduced. Injecting the `IActorRef` into the constructor of each query instead of having each query resolve the journal ref themselves makes it possible to throttle all queries launched by the same read journal plugin.